docs/design_proposals/generalizing_kaniko_config.md

```diff
@@ -0,0 +1,137 @@
+# Generalizing Container Registry config for Cluster builds
+
+* Author(s): @venkatk-25
+* Design Shepherd: @balopat
+* Date: 
+* Status: 
+
+## Background
+
+Kaniko supports 2 types of registries GCR and Docker, with former being mandatory always. 
+The implementation itself is repository specific and not extensible as we need to keep adding more logic for every new type of repository and keep maintaining compatibilty with registries/credhelpers.
+
+The current looks like below( I have kept only required pieces from original code snippet):
+``` go
+type ClusterDetails struct {
+       // GCR creds
+	PullSecret     string        `yaml:"pullSecret,omitempty"`
+	PullSecretName string        `yaml:"pullSecretName,omitempty"`
+
+	DockerConfig   *DockerConfig `yaml:"dockerConfig,omitempty"`
+}
+
+// DockerConfig contains information about the docker `config.json` to mount.
+type DockerConfig struct {
+	Path       string `yaml:"path,omitempty"`
+	SecretName string `yaml:"secretName,omitempty"`
+}
+```
```

Comment:
Can you add a section of related bugs in Background ?
- https://github.com/GoogleContainerTools/skaffold/issues/731

Reply:
Added the section for this